src/App.tsx

```diff
@@ -1,61 +1,157 @@
+import React, {
+  ChangeEvent, FormEvent, useState,
+} from 'react';
+import fetchedUsers from './api/users';
+import fetchedTodos from './api/todos';
+import { User } from './types';
+import { TodoList } from './components/TodoList';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const todosWithAssociatedUsers = fetchedTodos.map(todo => ({
+  ...todo,
+  user: fetchedUsers.find(user => user.id === todo.userId),
+}));
 
 export const App = () => {
+  const [todoItems, setTodoItems] = useState(todosWithAssociatedUsers);
+  const [formValues, setFormValues] = useState({ title: '', user: '' });
+  const [titleValidation, setTitleValidation] = useState({
+    value: '',
+    isValid: false,
+    displayError: false,
+  });
+
+  const [userValidation, setUserValidation] = useState({
+    value: '',
+    isValid: false,
+    displayError: false,
+  });
+
+  const handleTitleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value.trimStart();
+
+    setTitleValidation(prevValidation => ({
+      ...prevValidation,
+      displayError: !value,
+      isValid: !!value,
+      value,
+    }));
+
+    setFormValues(prevValues => ({
+      ...prevValues,
+      title: value,
+    }));
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setUserValidation(prevValidation => ({
+      ...prevValidation,
+      displayError: !value,
+      isValid: !!value,
+      value,
+    }));
+
+    setFormValues(prevValues => ({
+      ...prevValues,
+      user: value,
+    }));
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!titleValidation.isValid || !userValidation.isValid) {
+      setTitleValidation(prevValidation => ({
+        ...prevValidation,
+        displayError: !prevValidation.isValid,
+      }));
+      setUserValidation(prevValidation => ({
+        ...prevValidation,
+        displayError: !prevValidation.isValid,
+      }));
+
+      return;
+    }
+
+    const selectedUser = fetchedUsers.find(
+      user => user.name === formValues.user,
+    ) as User;
```

Comment:
that casting is not typesafe, you have to check if `selectedUser` is not undefined